internal/controller/managedcluster_controller.go

```diff
@@ -337,6 +338,42 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, managedCluster *h
 	return ctrl.Result{}, nil
 }
 
+func (r *ManagedClusterReconciler) needToRequeue(ctx context.Context, managedCluster *hmc.ManagedCluster) (bool, error) {
+	type objectToCheck struct {
+		gvr       schema.GroupVersionResource
+		condition string
+	}
+
+	var requeue bool
+	var errs error
+	for _, obj := range []objectToCheck{
+		{
+			gvr: schema.GroupVersionResource{
+				Group:    "cluster.x-k8s.io",
+				Version:  "v1beta1",
+				Resource: "clusters",
+			},
+			condition: "ControlPlaneInitialized",
```

Comment:
What about ControlPlaneReady and InfrastructureReady?

Reply:
InfrastructureReady is set to "ready" much sooner than ControlPlaneInitialized, so it is better to wait for the last event in list. Also ControlPlaneReady and InfrastructureReady don't show any messages, so it leads into displaying the combined message like ". . ." (so empty messages are concatenated with dots).

Reply:
I think it still makes sense to indicate when provisioning is stuck at the infrastructure level. If the Cluster conditions are not sufficient to understand the issue, we might consider extending our conditions to include the status of infrastructure objects later.

So both ControlPlaneReady and InfrastructureReady are still quite useful imo

>so it leads into displaying the combined message like ". . ." (so empty messages are concatenated with dots).

it should be easy to handle, also we don't really need to change cluster conditions imo